### ISSUE TITLE
Cant combine two lucene indexes in one query

### ISSUE THREAD

**Issue by vantonov1** [orientechnologies/orientdb#5868]
2.2.0-beta

Search with 2 lucene indexes like

 select from C1 where [p1,p2] LUCENE "p1:1 AND p2:2" 

where are 2 separate indexes for p1 and p2
produces orientechnologies.orient.core.exception.OCommandExecutionException: Cannot evaluate lucene condition without index configuration.

Create index for every combination of fields is not a solution
`

```
    final ODatabaseDocumentTx db_notx = new ODatabaseDocumentTx("memory:default").open("admin", "admin");
    final OrientVertexType c1 = new OrientGraphNoTx(db_notx).createVertexType("C1");               

    c1.createProperty("p1", OType.INTEGER);
    c1.createIndex("p1", "FULLTEXT", null, null, "LUCENE", new String[]{"p1"});

    c1.createProperty("p2", OType.INTEGER);
    c1.createIndex("p2", "FULLTEXT", null, null, "LUCENE", new String[]{"p2"});

    final ODatabaseDocumentTx db = new ODatabaseDocumentTx("memory:default").open("admin", "admin");
    final OrientGraph graph = new OrientGraph(db);
    graph.begin();
    final OrientVertex vertex = graph.addVertex("class:C1");
    vertex.setProperty("p1", 1);
    vertex.setProperty("p2", 1001);
    final List<?> result = db.command(new OSQLSynchQuery<>("select from C1 where [p1,p2] LUCENE \"(p1:1 AND p2:1001)\"")).execute();
    assert !result.isEmpty();
    graph.shutdown(true, true);
```

`

**Comment by luigidellaquila** [comment]
Hi @vantonov1 

this syntax is not supported for separate indexes. 

You have two alternatives here:

- use the indexes separately:
```
select from C1 where p1 LUCENE "1" AND p2 LUCENE "2"
```

- define a composite index
```
c1.createIndex("p1_p2", "FULLTEXT", null, null, "LUCENE", new String[]{"p1", "p2"});
```

Thanks

Luigi

**Comment by vantonov1** [comment]
Imagine p1 is from a superclass. Second solution means I need to include p1 into all indexes for all subclasses, that is not good

First solution is possible, but I need to control, where it is possible to do one query to lucene, and where I should combine it with orient syntax

IMHO, it is better to just put all fields marked as "lucene indexed" into one lucene index, not related to orient classes. I guess, it would be even better for performance - Lucene has its own caching, and it is more efficient to have common cache

**Comment by luigidellaquila** [comment]
Assigning to @robfrank, he is doing some tests on a "global" lucene index, maybe he can give you some details.

Thanks

Luigi

**Comment by robfrank** [comment]
@vantonov1 I'm working on a single big index per database, but logical partitioned by class. 
So you can define and index on multiple fields: 
```sql
create index Song.all on Song (title,author,lyrics) FULLTEXT ENGINE LUCENE
```
and the query it with a function:

```sql
select from <COMPLEX QUERY> where lucene_match( "Song.author:Fabbio Song.title:sun" ) = true 
```

note that in the lucene query you should specify className and fieldName.

I would like to hear from you some suggestion in form of hypothetical index definition, queries and results. Voice of users is very important for me at this point.
 Note that a global index without any class filter will return RID of different , uncorrelated classes: IMHO this couldn't be very useful.

**Comment by vantonov1** [comment]
For me it sounds great. One question is inheritance - definitely it is needed to find row by inherited fields

**Comment by robfrank** [comment]
The serach_class function is implemented in 3.0.x:
http://orientdb.com/docs/3.0.x/indexing/Full-Text-Index.html#searchclass
